programs/string_rotation.py

```diff
@@ -13,9 +13,11 @@ def string_rotation(s1, s2):
     for i in s2:
         l2.append(i)
 
-    for i in range(len(l2) - 1):
-        l2.append(l2[0])
-        l2.pop(0)
+    for i in range(len(l2) - 1):  # rotate l1 char by char
```

Comment:
This is a brute force solution. You have to be able to check this in O(n) time with no extra memory. 

Currently you are using O(n^2) time and O(n) memory